spec/tapioca/compilers/dsl/active_record_identity_cache_spec.rb

```diff
@@ -0,0 +1,218 @@
+          def self.fetch_by_title!(title, includes: nil); end
+
+          sig { params(index_values: T.untyped, includes: T.untyped).returns(T::Array[::Post]) }
+          def self.fetch_multi_by_blog_id(index_values, includes: nil); end
+
+          sig { params(index_values: T.untyped, includes: T.untyped).returns(T::Array[::Post]) }
+          def self.fetch_multi_by_title(index_values, includes: nil); end
+        end
+      RUBY
+      assert_equal(rbi_for(content), expected)
+    end
+
+    it("generates methods for combined cache_indexes") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          include IdentityCache
+          cache_index :title
+          cache_index :title, :review_date, unique: true
+        end
+
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { params(title: T.untyped, includes: T.untyped).returns(T::Array[::Post]) }
+          def self.fetch_by_title(title, includes: nil); end
+
+          sig { params(title: T.untyped, review_date: T.untyped, includes: T.untyped).returns(T.nilable(::Post)) }
+          def self.fetch_by_title_and_review_date(title, review_date, includes: nil); end
+
+          sig { params(title: T.untyped, review_date: T.untyped, includes: T.untyped).returns(::Post) }
+          def self.fetch_by_title_and_review_date!(title, review_date, includes: nil); end
+
+          sig { params(index_values: T.untyped, includes: T.untyped).returns(T::Array[::Post]) }
+          def self.fetch_multi_by_title(index_values, includes: nil); end
+        end
+      RUBY
+      assert_equal(rbi_for(content), expected)
+    end
+
+    it("generates methods for classes with cache_has_manys index") do
+      content = <<~RUBY
+        class User < ActiveRecord::Base
+        end
+
+        class Post < ActiveRecord::Base
+          include IdentityCache
+          has_many :users
+          cache_has_many :users
+        end
+
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T::Array[T.untyped]) }
+          def fetch_user_ids; end
+
+          sig { returns(T::Array[::User]) }
+          def fetch_users; end
+        end
+      RUBY
+      assert_equal(rbi_for(content), expected)
+    end
+
+    it("generates methods for classes with cache_has_one index") do
+      content = <<~RUBY
+        class User < ActiveRecord::Base
+        end
+
+        class Post < ActiveRecord::Base
+          include IdentityCache
+          has_one :user
+          cache_has_one :user, embed: :id
+        end
+
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(::User) }
+          def fetch_user; end
+
+          sig { returns(T.untyped) }
+          def fetch_user_id; end
+        end
+      RUBY
+      assert_equal(rbi_for(content), expected)
+    end
+
+    it("generates methods for classes with cache_belongs_to index") do
```

Comment:
This test is more testing for the polymorphic association, so it would be good to mention that in the name:
```suggestion
    it("generates methods for classes with cache_belongs_to index on a polymorphic relation") do
```